bob/src/link_manager.rs

```diff
@@ -21,28 +24,63 @@ impl LinkManager {
     }
 
     async fn checker_task(factory: Factory, nodes: Arc<[Node]>, period: Duration) {
+        let now = coarsetime::Clock::now_since_epoch().as_secs();
```

Comment:
Do not use `Clock` to measure time intervals. It is suffer from changes of the system clock. Use `Instant` + `Duration` instead